src/decode.rs

```diff
@@ -4588,8 +4678,8 @@ pub(crate) unsafe fn rav1d_decode_frame_init_cdf(
         let end: usize = tile.hdr.end.try_into().unwrap();
 
         let mut data = tile.data.as_ref();
-        for (j, (ts, tile_start_off)) in iter::zip(
-            slice::from_raw_parts_mut(f.ts, end + 1),
+        for (j, (ts_idx, tile_start_off)) in iter::zip(
+            0..end + 1,
             if uses_2pass {
                 &f.frame_thread.tile_start_off[..end + 1]
             } else {
```

Comment:
`j` is the same as `ts_idx` here, so we can simplify this a lot:

```rust
for (ts_idx, tile_start_off) in if uses_2pass {
    &f.frame_thread.tile_start_off
} else {
    &[]
}
.into_iter()
.map(|&it| it as usize)
.chain(iter::repeat(0))
.take(end + 1)
.enumerate()
.skip(start)
```

and then use `ts_idx` instead of `j`.

That said, why can't we keep the old code here?  We're still indexing into `f.ts` to get `ts` anyways.

Reply:
Ah, we can keep the original code, this was an accidental change that I had meant to revert. I had initially put the context data in a separate `DisjointMut` allocation to help with debugging, and forgot to revert this once I moved the data back into the `Rav1dTileState`. Reverted.

Reply:
Ah okay.